js/serial.js

```diff
@@ -12,10 +15,76 @@ var serial = {
     transmitting:   false,
     outputBuffer:  [],
 
+    LOGHEAD: 'SERIAL: ',
+
     connect: function (path, options, callback) {
         var self = this;
         self.openRequested = true;
 
+        var testUrl = path.match(/tcp:\/\/(.*):(.*)/)
+        if (testUrl) {
+            self.connectionIP = testUrl[1];
```

Comment:
I'd stick the two clauses of this if statement into `connectTcp` and `connectSerial` respectively to increase readability.